pkg/arvo/sys/lull.hoon

```diff
@@ -19,6 +19,412 @@
       size=@ud
       max-size=_64
   ==
+::
+::  +mop: constructs and validates ordered ordered map based on key,
+::  val, and comparator gate
+::
+++  mop
+  |*  [key=mold value=mold]
+  |=  ord=$-([key key] ?)
+  |=  a=*
+  =/  b  ;;((tree [key=key val=value]) a)
+  ?>  (apt:((on key value) ord) b)
+  b
+::
+::
+++  ordered-map  on
+::  +on: treap with user-specified horizontal order, ordered-map
+::
+::  WARNING: ordered-map will not work properly if two keys can be
+::  unequal under noun equality but equal via the compare gate
+::
+++  on
+  ~/  %on
```

Comment:
This is the kind-of hacky pattern we've historically used in %zuse:

```
~%  %on  ..part  ~
```

Reply:
Fixed in the other PR